src/lib/segmenta.ts

```diff
@@ -462,3 +452,39 @@ function looksLikeDSL(str: string): boolean {
     // assume dsl query if there are multiple words in the query string
     return parts.length > 1;
 }
+
+function prepareCachedResult(
+    options: ISanitizedQueryOptions,
+    cached: IPositionsResult
+) {
+    const
+        skip = options.skip || 0,
+        take = options.take === undefined ? cached.total : options.take,
```

Comment:
Minor: `take = options.take || cached.Total`

Reply:
It's quite legal, though of dubious usefulness, to set `take` to 0.

Perhaps the caller incorrectly calculates `take`, or is interested in the metadata. At any rate, I can't stamp over 0.